### DISCUSSION TITLE
How to change Line thickness and color?

### DISCUSSION THREAD

**Discussion by pwicks86** [py-pdf/pypdf#2503] [Q&A]
Is it possible to change the thickness and color of a line annotation?

**Comment by pwicks86** [comment]
Aha, figured out color:

```
line_annotation[NameObject("/C")] = ArrayObject([FloatObject(r), FloatObject(g), FloatObject(b)])
```

Still working on line width.

**Comment by pubpub-zz** [comment]
for width try:
![image](https://github.com/py-pdf/pypdf/assets/4083478/f0123eda-668d-4d85-a54d-93f0879175a2)

**Reply by R2RT** [reply]
For as for width this seem to work: (reverse engineered from Foxit Reader generated line)

```python
bs = DictionaryObject()
bs.update(
    {
        NameObject("/S"): NameObject("/S"),  # Solid border
        NameObject("/W"): FloatObject(1),  # Border width
    }
)
line_annotation[NameObject("/BS")] = bs
```

There is also option to set alpha channel:

```python
line_annotation[NameObject("/CA")] = FloatObject(0.8)  # alpha
```
